Examples/Scripts/Python/particle_gun.py

```diff
@@ -37,45 +78,65 @@ def runParticleGun(outputDir, s=None):
     g.particles = ptclGen
 
     evGen = EventGenerator(
-        level=acts.logging.INFO,
+        level=s.config.logLevel,
         generators=[g],
         outputParticles="particles_input",
         randomNumbers=rnd,
     )
 
     s.addReader(evGen)
 
-    s.addAlgorithm(
-        ParticlesPrinter(
-            level=acts.logging.INFO, inputParticles=evGen.config.outputParticles
+    if printParticles:
+        s.addAlgorithm(
+            ParticlesPrinter(
+                level=s.config.logLevel, inputParticles=evGen.config.outputParticles
+            )
         )
-    )
-
-    csv_dir = os.path.join(outputDir, "csv")
-    if not os.path.exists(csv_dir):
-        os.mkdir(csv_dir)
-
-    s.addWriter(
-        CsvParticleWriter(
-            level=acts.logging.INFO,
-            inputParticles=evGen.config.outputParticles,
-            outputDir=csv_dir,
-            outputStem="particles",
-        ),
-    )
 
-    root_file = os.path.join(outputDir, "particles.root")
+    if outputDirCsv is not None:
+        outputDirCsv = Path(outputDirCsv)
+        if not os.path.exists(outputDirCsv):
```

Comment:
Concerning your reply below: even if we allow `str` and `Path`, after the line that I marked here, it can only be `Path` right? In that case I would just use the `Path` API from that point on.

Reply:
Done